resource-recorder/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+use std::fmt;
+
+use async_trait::async_trait;
+use dal::{Dal, Resource};
+use prost_types::TimestampError;
+use shuttle_common::{backends::auth::VerifyClaim, claims::Scope};
+use shuttle_proto::resource_recorder::{
+    self, resource_recorder_server::ResourceRecorder, ProjectResourcesRequest, RecordRequest,
+    ResourcesResponse, ResultResponse, ServiceResourcesRequest,
+};
+use thiserror::Error;
+use tonic::{Request, Response, Status};
+
+pub mod args;
+mod dal;
+mod r#type;
+
+pub use dal::Sqlite;
+use tracing::error;
+use ulid::DecodeError;
+
+/// A wrapper to capture any error possible with this service
+#[derive(Error, Debug)]
+pub enum Error {
+    UlidDecode(#[from] DecodeError),
+    Dal(#[from] sqlx::Error),
+    String(String),
```

Comment:
Should it be like this?

```suggestion
    #[error("could not parse resource type: {0}")
    String(String),
```

Reply:
I'm implementing a custom `fmt::Display` to be able to hide sensitive `sqlx::Error` messages from users.